src/LeanCode.ContractsGenerator/Generation/TypeRefFactory.cs

```diff
@@ -62,6 +62,56 @@ public TypeRef From(ITypeSymbol symbol)
         }
     }
 
+    public NotificationTypeRef FromNotification(ITypeSymbol symbol)
+    {
+        return new()
+        {
+            Type = From(symbol),
+            Tag = GetFullTypeName(symbol),
+        };
+    }
+
+    private static string GetFullTypeName(ITypeSymbol typeSymbol)
```

Comment:
I'd like to see many unit tests for that function only. Since we want to imitate `typeof(T).ToString()` we already know the desired output, so having an xunit `Theory` with many types as arguments would be nice.

Also a comment about what that function tries to achieve would be nice.